src/test/lambda/config/parameterUtils.test.ts

```diff
@@ -0,0 +1,237 @@
+
+            const parameter = actual.get('MyParam')
+            assert.ok(parameter)
+            assert.strictEqual(parameter!.required, true)
+        })
+
+        it('sets `required` to false if default is defined, but falsy', async () => {
+            const context: GetParametersContext = {
+                loadTemplate: async () => ({
+                    Parameters: {
+                        MyParam: {
+                            Type: 'String',
+                            Default: false
+                        }
+                    }
+                })
+            }
+
+            const actual = await getParameters(vscode.Uri.file(''), context)
+            assert.strictEqual(actual.size, 1)
+
+            const parameter = actual.get('MyParam')
+            assert.ok(parameter)
+            assert.strictEqual(parameter!.required, false)
+        })
+
+        it('sets `required` to false if default is defined and truthy', async () => {
+            const context: GetParametersContext = {
+                loadTemplate: async () => ({
+                    Parameters: {
+                        MyParam: {
+                            Type: 'String',
+                            Default: true
+                        }
+                    }
+                })
+            }
+
+            const actual = await getParameters(vscode.Uri.file(''), context)
+            assert.strictEqual(actual.size, 1)
+
+            const parameter = actual.get('MyParam')
+            assert.ok(parameter)
+            assert.strictEqual(parameter!.required, false)
+        })
+    })
+
+    describe('getParameterNames', async () => {
+        it('returns an empty array if no parameters were found', async () => {
+            const context: GetParametersContext = {
+                loadTemplate: async () => ({
+                })
+            }
+
+            const actual = await getParameterNames(vscode.Uri.file(''), context)
+            assert.strictEqual(actual.length, 0)
+        })
+
+        it('returns the names of each parameter', async () => {
+            const context: GetParametersContext = {
+                loadTemplate: async () => ({
+                    Parameters: {
+                        MyParam1: {
+                            Type: 'String'
+                        },
+                        MyParam2: {
+                            Type: 'String'
+                        }
+                    }
+                })
+            }
+
+            const actual = await getParameterNames(vscode.Uri.file(''), context)
+            assert.strictEqual(actual.length, 2)
+            assert.strictEqual(actual[0], 'MyParam1')
+            assert.strictEqual(actual[1], 'MyParam2')
+        })
+    })
+
+    describe('getOverriddenParameters', async () => {
+        const workspaceFolderPath = path.join('my', 'workspace', 'folder')
+        const relativeTemplatePath = path.join('my', 'template.yaml')
```

Comment:
Minor: consider not using 'my' in the relative template path. On first glance it reads more absolute, because the workspace folder also starts with 'my'

Reply:
✔️